docs/spec/spec.md

```diff
@@ -0,0 +1,297 @@
+| Update | | |
+| Delete | | |
+
+---
+
+## Shared Object Schema
+
+### ProvisionerReference
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| ref<sup>1</sup> | ObjectReference | | |
+| selector<sup>1</sup> | LabelSelector | | |
+
+1: One of (name, selector), Required.
+
+### EndpointSpec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| targetRef<sup>1</sup> | ObjectReference | | Must adhere to Targetable. |
+| dnsName<sup>1</sup> | String | | |
+
+1: One of (targetRef, dnsName), Required.
+
+### ParameterSpec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| name*| String | The unique name of this template parameter. | |
+| description | String | A human-readable explanation of this template parameter. | |
+| required | Boolean | If the parameter is required. | If true, default / defaultFrom should not be provided. Immutable. |
+| default<sup>1</sup> | String | Default value if not provided by arguments. | |
+| defaultFrom<sup>1</sup> | ArgumentValueReference | Default value retrieved from an existing Secret or ConfigMap if not provided by arguments. | |
+
+\*: Required
+
+1: OneOf (default, defaultFrom)
+
+### ArgumentValueReference
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| secretKeyRef<sup>1</sup> | SecretReference | A reference to a value contained in a Secret at the given key. | |
+| configMapRef<sup>1</sup> | ConfigMapReference | A reference to a value contained in a ConfigMap at the given key. | |
+
+1: OneOf (secretKeyRef, configMapRef), Required.
+
+### ProvisionedObjectStatus
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| name*| String | Name of Object| |
+| type*| String | Fully Qualified Object type. | |
+| status*| String | Current relationship between Provisioner and Object| |
+| reason | String | Detailed description describing current relationship status. | |
+
+\*: Required
+
+### SubscriptionResolutionsStatus
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| call | String | Resolved target for the spec's call field. Empty string if spec.call is nil. | Must be a domain name |
+| continue | String | Resolved target for the spec's continue field. Empty string if spec.continue is nil. | Must be a domain name |
+
+### Subscribable
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| channelable | ObjectReference | The channel used to emit events. | |
+
+### Channelable
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| subscribers | ChannelSubscriberSpec[]| Information about subscriptions used to implement message forwarding. | Filled out by Subscription Controller. |
+
+### ChannelSubscriberSpec
+
+| Field | Type | Description | Limitations |
+| --- | --- | --- | --- |
+| callableDomain | String | The domain name of the endpoint for the call. | |
```

Comment:
There seem to be three different sets of names for these:

Subscription: call, result
SubscriptionResolutionStatus: call, continue
ChannelSubscriberSpec: Callable, Sinkable

Reply:
Thanks again! updated.